os/kernel/binary_manager/binary_manager_load.c

```diff
@@ -189,91 +176,53 @@ int binary_manager_load_binary(int bin_idx)
 		}
 	}
 
-	/* Failed to find valid binary */
-	if (loadable == false) {
-		bmdbg("Failed to find loadable binary %d\n", bin_idx);
+	if (valid_bin_count == 0) {
+		bmdbg("Failed to find valid header of binary %s\n", BIN_NAME(bin_idx));
 		return ERROR;
 	}
 
 	/* Load binary */
 	do {
-		if (is_new_bin == false) {
-			strncpy(load_attr.bin_name, header_data[latest_idx].bin_name, BIN_NAME_MAX);
-			snprintf(devname, BINMGR_DEVNAME_LEN, BINMGR_DEVNAME_FMT, BIN_PARTNUM(bin_idx, latest_idx));
-			load_attr.bin_size = header_data[latest_idx].bin_size;
-			load_attr.compression_type = header_data[latest_idx].compression_type;
-			load_attr.ram_size = header_data[latest_idx].bin_ramsize;
-			load_attr.stack_size = header_data[latest_idx].bin_stacksize;
-			load_attr.priority = header_data[latest_idx].bin_priority;
-			load_attr.offset = CHECKSUM_SIZE + header_data[latest_idx].header_size;		
-			is_new_bin = true;
-		}
-		bmvdbg("BIN[%d] %s %d %d\n", bin_idx, devname, load_attr.bin_size, load_attr.offset);
+		strncpy(load_attr.bin_name, header_data[latest_idx].bin_name, BIN_NAME_MAX);
```

Comment:
I think some of variable can be added new structure binary_header_attr_t also both of binary_header_t & load_attr_t can share it.
Then you can use memcpy easily.

Reply:
thank you. I will modify it later.